### PR TITLE
Upgrade to newer versions and fixed deprecations

### PR DESCRIPTION
**Pull request by s3tezsky**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Original package is not maintained and does not support Symfony 4. This PR solves deprecations and allow you to use the package with Symfony version 3 and 4
|New feature| No 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| Yes 
|Fixes issues| ...

### CODE REVIEW

File: tests/Builder/AbstractOrderedBuilderTest.php [hunk 1]
```diff
@@ -47,7 +48,7 @@ public function testDefaultState()
     }
 
     /**
-     * @expectedException \Symfony\Component\Form\Exception\BadMethodCallException
+     * @expectedException \BadMethodCallException
      * @expectedExceptionMessage The config builder cannot be modified anymore.
```
Comment (grossmannmartin) [inline tests/Builder/AbstractOrderedBuilderTest.php]: Why the expected exception is not in the code the same way as it is on line 95?

File: tests/Builder/AbstractOrderedBuilderTest.php [hunk 2]
```diff
@@ -47,7 +48,7 @@ public function testDefaultState()
     }
 
     /**
-     * @expectedException \Symfony\Component\Form\Exception\BadMethodCallException
+     * @expectedException \BadMethodCallException
```
Comment (grossmannmartin) [inline tests/Builder/AbstractOrderedBuilderTest.php]: The exception class is changed by you or by symfony?

File: tests/OrderedFormFunctionnalTest.php [hunk 3]
```diff
@@ -71,10 +73,9 @@ public function testInvalidPosition(array $config, $exceptionMessage = null)
     {
         $exceptionName = OrderedConfigurationException::class;
 
+        $this->expectException($exceptionName);
```
Comment (grossmannmartin) [inline tests/OrderedFormFunctionnalTest.php]: Can you inline these two lines? extra variable seems redundant to me

File: doc/docker.md [hunk 4]
```diff
@@ -38,12 +38,6 @@ To run the test suite, you can use:
 $ docker-compose run --rm php vendor/bin/phpunit
 ```
 
-If you want to run the test suite against [HHVM](http://hhvm.com/), you can use:
-
-``` bash
-$ docker-compose run --rm hhvm vendor/bin/phpunit
```
Comment (grossmannmartin) [inline doc/docker.md:44]: Why did you remove this? HHVM seems still present in a package.

**Review by grossmannmartin — CHANGES_REQUESTED** [review]
Hi @s3tezsky 
Awesome work is done here!
Can you please check the comments?

File: doc/docker.md (hunk 4 above)
Comment (s3tezsky) [reply doc/docker.md:44]: The travis build failed on HHVM and as we are not using it I did not want to bother with it.

I feel bad about removing it but it was the fastest way to deliver it.

File: tests/Builder/AbstractOrderedBuilderTest.php (hunk 1 above)
Comment (s3tezsky) [reply tests/Builder/AbstractOrderedBuilderTest.php]: For some reason the aannotation failed in other tests. That is why I move it inside the test. I will also move this one.

File: tests/Builder/AbstractOrderedBuilderTest.php (hunk 2 above)
Comment (s3tezsky) [reply tests/Builder/AbstractOrderedBuilderTest.php]: I cannot explain it, it is some mistake which should not be there. I will fix it.

File: doc/docker.md (hunk 4 above)
Comment (s3tezsky) [reply doc/docker.md:44]: As composer does not support HHVM anymore I will remove it all.

**Review by grossmannmartin — APPROVED** [review]
Please just add PR description. And I'm ok with it

**Review by TomasLudvik — APPROVED** [review]
